File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    can_reserve = 0
    
    real_lost = set(lost) - set(reserve)
    real_reserve = set(reserve) - set(lost)
    
    for i in sorted(real_lost):
        if i-1 in real_reserve:
            real_reserve.remove(i-1)
            can_reserve+=1
        elif i+1 in real_reserve:
            real_reserve.remove(i+1)
            can_reserve+=1
            
    return n - len(real_lost) + can_reserve

File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> total(n+1, 1);
    for(auto a:lost){
        total[a]--;
    }
    for(auto a:reserve){
        total[a]++;
    }
    for(int i=1; i<n+1; i++){
        if(total[i]==0){
            if(total[i-1]==2 && i>=2){
                total[i-1]--;
                total[i]++;
            }
            else if(total[i+1]==2 && i<=n-1){
                total[i+1]--;
                total[i]++;
            }
        }
    }
    for(auto i:total){
        if(i>=1){
            answer++;
        }
    }
    return answer-1;
}